stan/math/mix/functor/laplace_marginal_density.hpp

```diff
@@ -47,9 +48,12 @@ struct laplace_options {
   int max_num_steps{100};
 
   /* Initial value for theta. Defaults to 0s of the correct size if nullopt */
-  std::optional<Eigen::VectorXd> theta_0{std::nullopt};
+  Theta theta_0{0};
 };
 
+using laplace_options = laplace_options_base<Eigen::VectorXd>;
+using laplace_options_default = laplace_options_base<double>;
```

Comment:
I find stuffing a double in here that we never use to be pretty unfortunate. There are some [ways of making this compile away entirely](https://brevzin.github.io/c++/2021/11/21/conditional-members/#conditional-member-variables) but they're also kind of ugly. 

Could this be a case where using inheritance makes sense?

Reply:
Mkay I think I have a nicer approach. We use a base laplace options, then we have a `laplace_options` with a template type for `Theta` If theta is a vector we include the `Theta theta_0` member. Otherwise we don't include the member at all